ext/intl/calendar/calendar.stub.php

```diff
@@ -437,6 +437,10 @@ public function isSet(int $field): bool {}
      */
     public function set(int $year, int $month, int $dayOfMonth = UNKNOWN, int $hour = UNKNOWN, int $minute = UNKNOWN, int $second = UNKNOWN) {} // TODO make return type void
 
+    public function setDate(int $year, int $month, int $dayOfMonth): void {}
+
+    public function setDateTime(int $year, int $month, int $dayOfMonth, int $hour, int $minute, ?int $second = null): void {}
```

Comment:
Why is ``$second`` null? Surely making it default to ``0`` is more logical?

Reply:
It's because the underlying library code is overloaded, that's why `setDateTime` is implemented the following way:
```
if (second_is_null) {
    co->ucal->set((int32_t) year, (int32_t) month, (int32_t) day, (int32_t) hour, (int32_t) minute);
} else {
    ZEND_VALUE_ERROR_OUT_OF_BOUND_VALUE(second, 6);
    co->ucal->set((int32_t) year, (int32_t) month, (int32_t) day, (int32_t) hour, (int32_t) minute, (int32_t) second);
}
```

Reply:
I'm not sure though if the two signatures of `UCalendar::set` is really needed to be used, but I wanted to use the original code as much as possible.

Reply:
Right, but my point is that we can just use ``0`` seconds as a "sane" default, as what else is the behaviour? Not being displayed? If that's the case then some tests would be nice to see the difference between providing it and null.

Reply:
At last I managed to find a testcase which would fail if we always provided the 6th parameter:
```
$intlcal = IntlCalendar::createInstance('UTC');
$intlcal->clear();

$intlcal->setDateTime(2012, 1, 29, 23, 58);
var_dump($intlcal->isSet(IntlCalendar::FIELD_SECOND)); // currently, it's false but would be true with the suggested approach
var_dump($intlcal->get(IntlCalendar::FIELD_MINUTE));
var_dump($intlcal->isSet(IntlCalendar::FIELD_SECOND)); // it's true here... wtf :D
```

Reply:
ACK